src/hardware/siglent-sds/protocol.c

```diff
@@ -322,6 +322,7 @@ static int siglent_sds_read_header(struct sr_dev_inst *sdi)
 	ret = sr_scpi_read_data(scpi, buf, SIGLENT_HEADER_SIZE);
 	if (ret < SIGLENT_HEADER_SIZE) {
 		sr_err("Read error while reading data header.");
+		sr_dbg("Device returned %i bytes.", ret);
```

Comment:
This line was added when I was trying to debug why capturing a waveform wasn't working over USB. Doing this showed that it was only getting 52 bytes when done over USB. This line should hopefully help with future debugging.

Reply:
FYI see these findings on using USB on SDS1104X-E: https://github.com/sigrokproject/libsigrok/pull/114/files#diff-b2e40d0d92f027de6ccfdb0e8581aca4192f2408ec3eb900fcc1a6520d054fd3R39

The USB packet size is 64 bytes with a 12 byte packet header so you get 52 bytes of data per packet. Additionally at least the SDS1104X-E has an internal buffer of 61440 bytes (variable named show_send_buffer_size can be seen on uboot log) which gets refilled only when it has been completely read. And while it's refilling the read will fail.